flowz/test/artifacts/artifacts_test.py

```diff
@@ -0,0 +1,295 @@
+from collections import OrderedDict
+from concurrent import futures
+
+from nose import tools
+from tornado import gen
+from tornado import testing as tt
+import tornado.concurrent
+
+from flowz.artifacts import (ExtantArtifact, DerivedArtifact, ThreadedDerivedArtifact,
+                             WrappedArtifact, TransformedArtifact, KeyedArtifact,
+                             maybe_artifact)
+from flowz.channels import ChannelDone
+
+
+class ArtifactsTest(tt.AsyncTestCase):
```

Comment:
I would, but I'm assuming in so saying that this isn't a huge effort.  My sense is that a few more hours' work to make the tests more rugged is time well spent at this point.  But if you think I'm underestimating the effort, it could be a separate issue for a later time.